src/app/passengers/components/smart/trip-planner/trip-planner.component.ts

```diff
@@ -167,12 +171,61 @@ export class TripPlannerComponent implements OnInit {
     this.store.dispatch(new MapActions.AddLocation(map));
   }
 
+  getTripEstimate(event){
+    var res;
+
+    const headers = new HttpHeaders({
+      'Content-Type': 'application/x-www-form-urlencoded',
+      'Authorization': 'Basic ' + btoa('front-end:front-end')
+    });
+
+    const data = "grant_type=password&scope=webclient&username=passenger&password=password";
+
+    const options = {
+      headers,
+      withCredentials: true
+    }; 
+
+   this.http.post('http://localhost:32775/auth/oauth/token', data, options).subscribe(res => {
```

Comment:
Because the port is subject to change, do you mind extracting the port or hostname as a variable, so future developers don't have to search the codebase

Reply:
port number has been made configurable